src/mots/bmo.py

```diff
@@ -0,0 +1,108 @@
+"""Module that provides helpers to interact with the Bugzilla API."""
+
+# TODO: this should be abstracted so that it can be hot-swappable with any API that can
+# be searched.
+
+from __future__ import annotations
+import requests
+import os
+import logging
+import time
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_BASE_URL = "https://bugzilla.mozilla.org/rest"
+USER_AGENT = "mots"  # TODO: improve this and include version.
+
+# TODO set NO_INPUT var to prevent prompt.
+
+
+class BMOClient:
+    """A thin wrapper as a Bugzilla API client."""
+
+    def __init__(self, token: str = None, base_url: str = DEFAULT_BASE_URL):
+        if not token:
+            token = os.getenv("BUGZILLA_API_KEY", "")
+            if not token:
+                token = input("Enter BMO Token: ")
+            if not token:
+                raise Exception()
```

Comment:
I'm -1 on using `input` here. If we really need to use it, it should probably be a level above in the CLI.

We should add some more useful information to that exception, and maybe make it a `ValueError`.